dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -1504,16 +1512,23 @@ private boolean resolveHang(long tasksCompleted, boolean queueEmpty, int poolSiz
                 Tr.event(tc, "Executor hang continued at poolSize=" + poolSize, threadPool);
             }
 
-            setPoolIncrementDecrement(poolSize);
-            if (poolSize + poolIncrement <= maxThreads && poolSize < MAX_THREADS_TO_BREAK_HANG) {
-                targetPoolSize = adjustPoolSize(poolSize, poolIncrement);
+            /**
+             * We might get here with a prior pool size change incomplete ... if so we
+             * should adjust for that
+             */
+            int hangSizeAdjustment = 0;
+            if (poolSize > targetPoolSize) {
+                setPoolIncrementDecrement(targetPoolSize);
+                hangSizeAdjustment = poolIncrement - (poolSize - targetPoolSize);
+            } else if (poolSize < targetPoolSize) {
+                hangSizeAdjustment = targetPoolSize - poolSize;
+            }
```

Comment:
This is relatively doing the same logic as checkTargetPoolSize.  The call to checkTargetPoolSize is done right after resolveHang.  Should we change to call that here instead so that the other code get run that is in the checkTargetPoolSize.  At least some logic possibly should be updated so that we reset consecutiveTargetPoolSizeWrong if it isn't wrong and previously it was wrong.